src/general/base_stage/CreatureStageHUDBase.cs

```diff
@@ -665,6 +671,19 @@ protected virtual void UpdateHealth(float delta)
         var hpText = StringUtils.FormatNumber(Mathf.Round(hp)) + " / " + StringUtils.FormatNumber(maxHP);
         hpLabel.Text = hpText;
         hpLabel.TooltipText = hpText;
+
+        var shaderMaterial = (ShaderMaterial)damageScreenEffect.Material;
+
+        if ((float)shaderMaterial.GetShaderParameter(fadeParameterName) > 0)
```

Comment:
Shader parameters shouldn't be read back as they are much more performance intensive than just having a field in this class that keeps track of the last value applied to the shader.